### PR TITLE
chore: add wait-on-circle-workflows polling script for release automation

### LINKED ISSUE #25160: Write “Test Workflows Passed” script to poll when all other CircleCI Workflows have succeeded
@emilyrohrbough commented on [Tue Sep 20 2022](https://github.com/cypress-io/cypress-internal/issues/487)

- Write “Test Workflows Passed” script to poll when all other CircleCI Workflows have succeeded.
    - To use in App Release Workflow

### PR DESCRIPTION
**Pull request by emilyrohrbough**
Ensures all pipeline workflows have been triggered and have passed when the `develop` branch is built. This acts as the "gate-keeper" for moving forward with a release once all workflows are successful.

- Closes #25160

### CODE REVIEW

File: scripts/wait-on-circle-workflows.js [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+  _.remove(workflows, (w) => w.name === 'linux-x64') // this is the workflow that is running this job
+
+  console.log('workflows', workflows)
+
+  const missingWorkflows = WORKFLOW_NAMES.filter((w) => !_.find(workflows, { name: w }))
+
+  if (missingWorkflows.length) {
+    console.error('The following', missingWorkflows.length, 'workflows are required to release and have not been started:\n -', missingWorkflows.join('\n - '))
+    console.error('Fail early rather than wait for pipelines to finish.')
```
Comment (ryanthemanuel) [inline scripts/wait-on-circle-workflows.js]:
```suggestion
    console.error('Failing early rather than wait for pipelines to finish.')
```

**Comment by BlueWinds** [comment]
I'm not clear how this ties in with the rest of the CI pipeline.

I see that the job waits on a bunch of things to be true then logs "ready to release" - but not how this ties in with actually triggering the release? Because nothing seems to wait for trigger-binary-release-workflow in turn. Can you walk me through how this is supposed to work?

File: .circleci/workflows.yml [hunk 2]
```diff
@@ -2557,6 +2580,59 @@ linux-x64-workflow: &linux-x64-workflow
     # This release definition must be updated with any new jobs
     # Any attempts to automate this are welcome
     # If CircleCI provided an "after all" hook, then this wouldn't be necessary
```
Comment (mjhenkes) [inline .circleci/workflows.yml:2582]: > If CircleCI provided an "after all" hook, then this wouldn't be necessary


That would be nice

**Comment by emilyrohrbough** [comment]
@BlueWinds This ensures the  `darwin-arm64`, `linux-arm64`, and `windows` workflows have passed and all `darwin-x64` jobs until this point have passed before allowing the release script to run.

> Because nothing seems to wait for trigger-binary-release-workflow in turn. 

The release scripts / CI jobs to run this has not been implemented yet which is why this looks like it does nothing. We could prob comment this out at this point. This work will be tackled with #25159 & #25161

I didn't comment it out originally because if things seem wonky before then, we can iron that out before we completely rely on this.

**Review by mjhenkes — APPROVED** [review]

**Review by ryanthemanuel — APPROVED** [review]

**Pull request merged**